internal/translation/teams/conversion.go

```diff
@@ -0,0 +1,131 @@
+package teams
+
+import (
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type Team struct {
+	Roles    []string
+	TeamName string
+	TeamID   string
+}
+
+type AssignedTeam struct {
+	Usernames []string
+	TeamName  string
+	TeamID    string
+}
+
+type TeamUser struct {
+	Username string
+	UserID   string
+}
+
+func NewTeam(projTeamSpec *akov2.Team, teamID string) *Team {
+	if projTeamSpec == nil {
+		return nil
+	}
+
+	roles := make([]string, 0)
+	for _, role := range projTeamSpec.Roles {
+		roles = append(roles, string(role))
+	}
+
+	team := &Team{
+		Roles:  roles,
+		TeamID: teamID,
+	}
+
+	return team
+}
+
+func NewAssignedTeam(teamSpec *akov2.TeamSpec, teamID string) *AssignedTeam {
+	if teamSpec == nil {
+		return nil
+	}
+	usernames := make([]string, 0)
+	for _, username := range teamSpec.Usernames {
+		usernames = append(usernames, string(username))
+	}
+
+	team := &AssignedTeam{
+		TeamID:    teamID,
+		TeamName:  teamSpec.Name,
+		Usernames: usernames,
+	}
+
+	return team
+}
+
+func TeamFromAtlas(team *admin.TeamResponse) *Team {
+	if team == nil {
+		return nil
+	}
+	tm := &Team{
+		TeamID:   team.GetId(),
+		TeamName: team.GetName(),
+	}
+	return tm
+}
+
+func TeamToAtlas(team *akov2.Team, teamID string) *Team {
+	roleNames := make([]string, 0)
+	for _, role := range team.Roles {
+		roleNames = append(roleNames, string(role))
+	}
+	return &Team{
+		TeamID: teamID,
+		Roles:  roleNames,
+	}
```

Comment:
The role names conversion could be delegated so that conversions look more regular.

Also does it make sense to return nil if the input team is nil here? I am not sure myself, just asking

```suggestion
	if team == nil {
	  return nil
	}
	return &Team{
		TeamID: teamID,
		Roles:  RolesToAtlas(team.Roles),
	}
```

Reply:
Noted, will add! And I think a nil check is a good idea, I'll add this to the other ToAtlas functions as well